DDD.Core/AzureStorage/PresenterEntity.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using DDD.Core.Domain;
+using DDD.Core.Time;
+using Microsoft.WindowsAzure.Storage.Table;
+using Newtonsoft.Json;
+
+namespace DDD.Core.AzureStorage
+{
+    public class PresenterEntity : TableEntity
+    {
+        public PresenterEntity() { }
+
+        public PresenterEntity(Presenter presenter, string conferenceInstance)
+        {
+            PartitionKey = conferenceInstance;
+            RowKey = presenter.Id.ToString();
+            ExternalId = presenter.ExternalId;
+            Name = presenter.Name;
+            Presenter = JsonConvert.SerializeObject(presenter);
+        }
+
+        public Presenter GetPresenter()
+        {
+            return JsonConvert.DeserializeObject<Presenter>(Presenter);
+        }
+
+        public Guid Id => Guid.Parse(RowKey);
+
+        public string ExternalId { get; set; }
+        public string Name { get; set; }
+        public string Presenter { get; set; }
+
+        public PresenterEntity Update(Presenter newData, IDateTimeProvider dateTimeProvider)
+        {
+            var existing = GetPresenter();
+            newData.UpdateFromExisting(existing, dateTimeProvider);
```

Comment:
In the constructor we set the RowKey to the Id of the supplied presenter instance. When we update the presenter details we do not check if it's the same presenter as the existing one. Feels like we should reject if this is not the case.

Reply:
Perhaps it's premature optimization but I would consider holding a reference to the original Presenter instance rather than just the JSON string and then having to deserialize it again.

Reply:
Unsure what you mean by holdin the instance? I have to have it as a string for it to serialize correctly into table storage unfortunately.